backend/app/controllers/application_controller.rb

```diff
@@ -20,4 +20,12 @@ def authenticate!
   rescue JWT::ExpiredSignature
     head :unauthorized
   end
+
+  def set_locale
+    I18n.locale = extract_locale_from_accept_language_header
```

Comment:
In order to make sure we will not break the app with weird/wrong values from header we could fallback to default:

```ruby
  def set_locale
    I18n.locale = valid_locale || I18n.default_locale
  end

  def valid_locale
    locale = request.env['HTTP_ACCEPT_LANGUAGE'].to_s.scan(/^[a-z]{2}/).first
    locale if I18n.available_locales.map(&:to_s).include?(locale)
  end
```